### PR TITLE
[1222] Support add edit accrediting provider Step 1 non JS

### PR DESCRIPTION
**Pull request by darokel**
### Context

https://trello.com/c/arBbJOAy/1222-support-add-edit-accrediting-provider-step-1-non-js

https://user-images.githubusercontent.com/616080/233372170-7793af72-4bc1-4e3a-bca1-7bc5fa03455a.mp4

### Changes proposed in this pull request

- Extracts and re-uses logic for school search to enable it for searching accredited providers
- Moved some files around outside of namespaces as they will be used by both support and providers

### Guidance to review

- Login into support console and view a lead school
- Search for an AP using different terms (as specified in the requirements)

The submission just takes you back to the index page as this PR focuses on getting the search plumbing in. Next set of tickets will handle the rest of the flow.

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased main
- [x] Cleaned commit history
- [x] Tested by running locally
- [x] Inform data insights team due to database changes

### CODE REVIEW

**Comment by darokel** [comment]
[Support: Add & Edit Accrediting Provider - Step 1 (Non JS)](https://trello.com/c/arBbJOAy/1222-support-add-edit-accrediting-provider-step-1-non-js)

File: app/views/support/providers/accredited_provider_search/results.html.erb [hunk 1]
```diff
@@ -0,0 +1,40 @@
+
+<% content_for :page_title, search_result_title_component.title %>
```
Comment (defong) [inline app/views/support/providers/accredited_provider_search/results.html.erb:2]: Missing error prefix?

File: app/views/support/providers/users/index.html.erb [hunk 2]
```diff
@@ -1,3 +1,3 @@
-<%= link_to t("components.page_titles.support.providers.users.new"), new_support_recruitment_cycle_provider_user_path(@provider.recruitment_cycle_year, @provider), class: "govuk-!-margin-top-4 govuk-button" %>
+<%= link_to t("components.page_titles.support.providers.users.new"), new_support_recruitment_cycle_provider_user_path(@provider.recruitment_cycle_year, @provider), class: "govuk-button" %>
```
Comment (defong) [inline app/views/support/providers/users/index.html.erb:1]: maybe

https://govuk-components.netlify.app/helpers/button/#link-styled-like-a-button

File: app/helpers/provider_helper.rb [hunk 3]
```diff
@@ -44,6 +44,10 @@ def visa_sponsorship_short_status(provider)
     end
   end
 
+  def provider_short_address(provider)
+    [provider.address3, provider.postcode].select(&:present?).join(', ')
```
[the remark this answers is not in the document]
Comment (darokel) [reply app/helpers/provider_helper.rb:48]: Nothing it's gone

File: app/views/support/providers/accredited_provider_search/results.html.erb (hunk 1 above)
Comment (darokel) [reply app/views/support/providers/accredited_provider_search/results.html.erb]: done

File: app/components/search_result_title_component.rb [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  end
+
+  def query_text
+    return "for ‘#{query}’" if query.present?
+
+    'for your search'
+  end
+
+  def many_results_text
+    t('.many_results_html', link: govuk_link_to('Try narrowing down your search', return_path))
```
[the remark this answers is not in the document]
Comment (darokel) [reply app/components/search_result_title_component.rb]: not sure what you mean 🤔

File: spec/components/search_result_title_preview.rb [hunk 5]
```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+class SearchResultTitlePreview < ViewComponent::Preview
+  def with_many_results = render_component(10_000)
+
+  def with_few_results = render_component(10)
+
+  def with_1_result = render_component(1)
+
+  def with_no_results = render_component(0)
```
[the remark this answers is not in the document]
Comment (darokel) [reply spec/components/search_result_title_preview.rb]: everything works

**Review by simonwhatley — APPROVED** [review]

File: app/components/search_result_title_component.rb (hunk 4 above)
[the remark this answers is not in the document]
Comment (defong) [reply app/components/search_result_title_component.rb]: I meant doing that on the `en.yml`


```
    t('.many_results_html', results_limit:, link: govuk_link_to('Try narrowing down your search', return_path))
```
```
Showing the first %{results_limit} results. %{link} if the %{search_resource} you're looking for is not listed.
```

**Review by defong — APPROVED** [review]

File: app/components/search_result_title_component.rb (hunk 4 above)
[the remark this answers is not in the document]
Comment (darokel) [reply app/components/search_result_title_component.rb]: ah ok done

**Pull request merged**